frameworks/cassandra/tests/test_tls.py

```diff
@@ -0,0 +1,256 @@
+        ; This is disabled by default on all Instaclustr-managed clusters.
+        ; userkey = /path/to/userkey.pem
+        ; usercert = /path/to/usercert.pem
+        """.format(
+            hostname=hostname,
+            port=port,
+            certfile=certfile
+        ))
+
+
+def get_write_data_job(dcos_ca_bundle: str):
+    query = ' '.join([
+        "CREATE KEYSPACE testspace1 WITH REPLICATION = { 'class' : 'SimpleStrategy', 'replication_factor' : 3 };",
+        "USE testspace1;",
+        "CREATE TABLE testtable1 (key varchar, value varchar, PRIMARY KEY(key));",
+        "INSERT INTO testspace1.testtable1(key, value) VALUES('testkey1', 'testvalue1');",
+
+        "CREATE KEYSPACE testspace2 WITH REPLICATION = { 'class' : 'SimpleStrategy', 'replication_factor' : 3 };",
+        "USE testspace2;",
+        "CREATE TABLE testtable2 (key varchar, value varchar, PRIMARY KEY(key));",
+        "INSERT INTO testspace2.testtable2(key, value) VALUES('testkey2', 'testvalue2');"])
+    return _get_cqlsh_job_over_tls(
+        'write-data',
+        [_get_cqlsh_for_query(query)],
+        dcos_ca_bundle=dcos_ca_bundle,
+    )
+
+
+def get_verify_data_job(dcos_ca_bundle: str):
+    commands = [
+        '{command} | grep testkey1'.format(command=_get_cqlsh_for_query('SELECT * FROM testspace1.testtable1;')),
+        '{command} | grep testkey2'.format(command=_get_cqlsh_for_query('SELECT * FROM testspace2.testtable2;')),
+    ]
+    return _get_cqlsh_job_over_tls(
+        'verify-data', commands, dcos_ca_bundle=dcos_ca_bundle)
+
+
+def get_delete_data_job(dcos_ca_bundle: str):
+    query = ' '.join([
+        'TRUNCATE testspace1.testtable1;',
+        'TRUNCATE testspace2.testtable2;',
+        'DROP KEYSPACE testspace1;',
+        'DROP KEYSPACE testspace2;'])
+    return _get_cqlsh_job_over_tls(
+        'delete-data',
+        [_get_cqlsh_for_query(query)],
+        dcos_ca_bundle=dcos_ca_bundle,
+    )
+
+
+def get_verify_deletion_job(dcos_ca_bundle: str):
+    commands = [
+        '{command} | grep "0 rows"'.format(
+            command=_get_cqlsh_for_query('SELECT * FROM system_schema.tables WHERE keyspace_name=\'testspace1\';')),
+        '{command} | grep "0 rows'.format(
+            command=_get_cqlsh_for_query('SELECT * FROM system_schema.tables WHERE keyspace_name=\'testspace2\';')),
+    ]
+    return _get_cqlsh_job_over_tls(
+        'delete-data', commands, dcos_ca_bundle=dcos_ca_bundle)
+
+
+def _get_cqlsh_job_over_tls(
+        name: str,
+        commands: List[str],
+        dcos_ca_bundle: str=None):
+    """
+    Creates a DC/OS job with `cqlsh` utility that will be ready to run commands
+    over a TLS connection.
+    """
+    commands_with_tls_prepare = [
+        'echo -n "$CQLSHRC_FILE" > $MESOS_SANDBOX/cqlshrc',
+        'echo -n "$CA_BUNDLE" > $MESOS_SANDBOX/ca-bundle.crt',
+    ]
+    commands_with_tls_prepare.extend(commands)
+    cmd = ' && '.join(commands_with_tls_prepare)
+
+    job = {
```

Comment:
Would it make sense to call `config. _get_test_job()` here and add the security requirements afterwards instead? I see that in #1495 we add a retry on failure, and it would be good to keep things consistent.